sdk/containerregistry/Azure.Containers.ContainerRegistry/TROUBLESHOOTING.md

```diff
@@ -160,4 +163,16 @@ The `CredentialUnavailableExcpetion` is a special exception type derived from `A
 
 Calls to service clients resulting in `RequestFailedException` with a `StatusCode` of 401 or 403 often indicate the caller doesn't have sufficient permissions for the specified API. Check the service documentation to determine which RBAC roles are needed for the specific request, and ensure the authenticated user or service principal have been granted the appropriate roles on the resource.
 
-For more help with troubleshooting authentication errors, see the Azure Identity client library [troubleshooting guide](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/TROUBLESHOOTING.md).
\ No newline at end of file
+For more help with troubleshooting authentication errors, see the Azure Identity client library [troubleshooting guide](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/TROUBLESHOOTING.md).
+
+## Service errors
+
+When working with `ContainerRegistryContentClient`, you may get an `RequestFailedException` exception with
+message containing additional information and [Docker error code](https://docs.docker.com/registry/spec/api/#errors-2).
+
+### Getting BLOB_UPLOAD_INVALID
+
+In rare cases, a transient error (such as connection reset) can happen during blob upload which may lead to a `RequestFailedException` with a 404 status code being thrown with message similar to
+`{"errors":[{"code":"BLOB_UPLOAD_INVALID","message":"blob upload invalid"}]}`, and resulting in a failed upload.  In this case, upload should to be restarted from the beginning.
+
+To display detailed error messages, please see the code sample in [General troubleshooting](#general-troubleshooting) above.
```

Comment:
nit: do we provide access to [`AcrErrors`](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerregistry/data-plane/Azure.ContainerRegistry/stable/2021-07-01/containerregistry.json#L2121) ? 

Assuming I'm a user and implementing restart logic in my custom code, I would not want to rely on exception message to retry call to service.
And also I don't want to retry all 404s from upload (e.g. if the repository does not exist), so is there a way to  get a code (`BLOB_UPLOAD_INVALID`) that users can retry upon?

here's what I did in JAva
```java
  if (ex.getCause() instanceof AcrErrorsException) {
      AcrErrorsException acrErrors = (AcrErrorsException) ex.getCause();
      for (AcrErrorInfo info : acrErrors.getValue().getErrors()) {
          System.out.printf("Uploaded blob failed: code '%s'\n", info.getCode());
      }
  }
```

Reply:
.NET doesn't provide service-specific exceptions (that I know of), but you're correct that this is a rare instance where the caller would want to programmatically take action on the exception type.  I can update the sample to show how to retrieve the error code to check it in order to determine a retry case.

Reply:
Discussed offline.